File: client/db-coursework/src/app/_models/orderDto.ts
import { DeliveryInfo } from "./delivery-info.model";
import { OrderItem } from "./order-item.model";

export class OrderDto {
    city: string;
    address: string;
    postalCode: string;
    carrier: string;
    cartItems: OrderItem[];

    constructor(deliveryInfo: DeliveryInfo, orderItems: OrderItem[]) {
        this.city = deliveryInfo.city;
        this.address = deliveryInfo.address;
        this.postalCode = deliveryInfo.postalCode;
        this.carrier = deliveryInfo.carrier;
        this.cartItems = orderItems;
    }
}

File: DB_Coursework_API/Models/DTO/OrderDto.cs
namespace DB_Coursework_API.Models.DTO
{
    public class OrderDto
    {
        public string City { get; set; }
        public string Address { get; set; }
        public string PostalCode { get; set; }
        public CartItem[] CartItems { get; set; }
    }
}
